src/libraries/System.Private.CoreLib/src/System/Double.cs

```diff
@@ -440,6 +440,162 @@ public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatPro
+
+        // TODO: make the helpers saturating when implementing https://github.com/dotnet/runtime/issues/61885
+        private static int ConvertToInt32(double val) => (int)(long)val;
+
+        private static ulong ConvertToUInt64(double val)
+        {
+            const double two63 = IntMaxValueOffset * UIntMaxValueOffset;
+            ulong ret;
+            // don't remove the double casts, the runtime would call this method recursively without them
+            if (val < two63)
+            {
+                ret = (ulong)(long)val;
+            }
+            else
+            {
+                // subtract 0x8000000000000000, do the convert then add it back again
+                ret = (ulong)(long)(val - two63) + 0x8000000000000000UL;
+            }
+            return ret;
+        }
+
+        [StackTraceHidden]
+        private static int ConvertToInt32Overflow(double val)
+        {
+            // Note that this expression also works properly for val = NaN case
+            if (val is > -IntMaxValueOffset - 1 and < IntMaxValueOffset)
+            {
+                int ret = (int)val;
+                // since no overflow can occur, the value always has to be within 1
+                Debug.Assert(val - 1.0 <= ret);
+                Debug.Assert(ret <= val + 1.0);
+                return ret;
+            }
+
+            ThrowHelper.ThrowOverflowException();
+            return 0;
+        }
+
+        [StackTraceHidden]
+        private static uint ConvertToUInt32Overflow(double val)
+        {
+            // Note that this expression also works properly for val = NaN case
+            if (val is > -1.0 and < UIntMaxValueOffset)
+            {
+                uint ret = (uint)val;
+                // since no overflow can occur, the value always has to be within 1
+                Debug.Assert(val - 1.0 <= ret);
+                Debug.Assert(ret <= val + 1.0);
+                return ret;
+            }
+
+            ThrowHelper.ThrowOverflowException();
+            return 0;
+        }
+
+        [StackTraceHidden]
+        private static long ConvertToInt64Overflow(double val)
+        {
+            const double two63 = IntMaxValueOffset * UIntMaxValueOffset;
+
+            // Note that this expression also works properly for val = NaN case
+            // We need to compare with the very next double to two63. 0x402 is epsilon to get us there.
+            if (val is > -two63 - 0x402 and < two63)
+            {
+                long ret = (long)val;
+                // since no overflow can occur, the value always has to be within 1
+                Debug.Assert(val - 1.0 <= ret);
+                Debug.Assert(ret <= val + 1.0);
+                return ret;
+            }
+
+            ThrowHelper.ThrowOverflowException();
+            return 0;
+        }
+
+        [StackTraceHidden]
+        private static ulong ConvertToUInt64Overflow(double val)
```

Comment:
If we are aligning with the public surface naming conventions, these methods should have `Checked` suffix (prior art: `TryConvertFromChecked`,  `CreateChecked`, ...)

Reply:
Done.